### DISCUSSION TITLE
USAGE.md vs. sample code

### DISCUSSION THREAD

**Discussion by andregasser** [andregasser/bigbone#64] [General]
I see some redundancy in maintaining a USAGE.md and our `sample` and `sample-kotlin`collection. What if we continue to just maintain our collection of samples and drop USAGE.md? The samples could also help to test individual cases and might be more attractive to developers that want to use certain features.

**Comment by bocops** [comment]
That sounds like a good idea. Having our example code in actual, well-documented code files will also help with ensuring that the example aren't outdated while we change things elsewhere. That has obviously already been a problem in the past.

**Reply by andregasser** [reply]
Great! 😀 Then let's move forward like this. I propose to keep the samples small (covering a single use case each). Ideally, we cover use cases that are used by developers, so they can get started quickly. Thanks for your feedback!
